File: Day 1/by.js
function countChar(words, char) {
    let count = 0;
    let positions = {};
    for (let word of words) {
        positions[word] = [];
        for (let i = 0; i < word.length; i++) {
            if (word[i] === char) {
                positions[word].push(i);
                count++;
            }
        }
    }
    return [count, positions];
}

let words = prompt("Enter words separated by space: ").split(" ");
let char = prompt("Enter a character: ");
let [count, positions] = countChar(words, char);
console.log(`'${char}' appears ${count} times.`);
console.log("Positions in each word:", positions);

File: Day 1/by.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Program
{
    public static void Main()
    {
        Console.WriteLine("Enter words separated by space: ");
        var words = Console.ReadLine().Split(' ');
        Console.WriteLine("Enter a character: ");
        var c = Console.ReadLine()[0];

        var count = 0;
        var positions = new Dictionary<string, List<int>>();
        foreach (var word in words)
        {
            var indices = new List<int>();
            for (var i = 0; i < word.Length; i++)
            {
                if (word[i] == c)
                {
                    count++;
                    indices.Add(i);
                }
            }
            positions[word] = indices;
        }

        Console.WriteLine($"'{c}' appears {count} times.");
        Console.WriteLine("Positions in each word:");
        foreach (var kvp in positions)
        {
            Console.WriteLine($"{kvp.Key}: [{string.Join(", ", kvp.Value)}]");
        }
    }
}
